settings.gradle

```diff
@@ -45,3 +45,20 @@ rootProject.children.each {
     it.name = it.name + "_" + scalaVersion
   }
 }
+
+/**
+ * Skips execution of all tasks for the project 'samza-test'.
+ * To include `samza-test`, initiate gradle build by setting the property 'includeSamzaTest'.
+ * To build with samza-test: './gradlew clean build -PincludeSamzaTest'
+ */
+def propertyName = "includeSamzaTest"
+def projectName = "samza-test"
+gradle.taskGraph.whenReady { taskGraph ->
+  taskGraph.getAllTasks().each { task ->
+    def project = task.getProject()
+    task.onlyIf {
+      // Closure, returns false for samza-test(will get filtered) and true for all other modules(will remain unfiltered).
+      project.hasProperty(propertyName) || !project.getName().contains(projectName)
```

Comment:
How does the first condition `project.hasProperty(propertyName)` work? Is the property inherited by every project when set? If so, preceding comment is incorrect/incomplete.